### PR TITLE
Deep linking set-up for page subsections

### PR DESCRIPTION
**Pull request by taysea**
This PR creates a proposal for how we can set up deep linking on the site for subsections. For now, I've just applied it to the 'About' page under the 'Start' item in the main nav. Now, each subsection receives a 'name' prop which applies the SubHeading and creates the id for the section. If we are on the same page with this approach, I can adjust the other files. For the color of the link icon/hover interaction, I've based it slightly off of grommet site and just chose a color from the new theme, but we should ask Chris about his thoughts on this design piece.

### CODE REVIEW

File: aries-site/src/layouts/content/Subsection.js [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+          gap="xsmall"
+          onMouseOver={() => setOver(true)}
+          onMouseOut={() => setOver(false)}
+          onFocus={() => setOver(true)}
+          onBlur={() => setOver(false)}
+        >
+          <Subheading>{name}</Subheading>
+          <Anchor
+            href={`#${id}`}
+            icon={<LinkIcon color={over ? 'text-xweak' : 'background'} />}
```
[the remark this answers is not in the document]
Comment (taysea) [reply aries-site/src/layouts/content/Subsection.js]: Yes, I think transparent is a good alternative.

[the remark this answers is not in the document]
Comment (taysea) [reply aries-site/src/layouts/content/Subsection.js]:
```suggestion
            icon={<LinkIcon color={over ? 'text-xweak' : 'transparent'} />}
```

**Comment by taysea** [comment]
Thanks for the feedback, Matt!

1. I think that this could live at the level of `Subheading` as opposed to automatically integrating it into the `Subsection`. 

2. I think we could apply the same logic to h3s if we end up having those -- I think as of now we just have h1 and h2 for the pages we've put content for. But this could be extended to h3s.

3. On grommet-site, the deep-linking tags items as `property.name`, so that's where I'd pulled `name` from but I am not opposed to using label.

4. The reason I hadn't thought of doing one for h1 (MainHeading) is because there's only going to be one h1 per page, so linking someone to myurl.com/start maybe doesn't need something like myurl.com/start#start. However, I do see the use case if we had a hero component and that heading gets pushed down. Considering these heading components are specific to aries-site, I think we should only add deeplinking to h1 if there is a specific design that necessitates that. From Chris's designs so far though, it seems like the h1 also sits right at the top so it might not be necessary but definitely open to discussing this.

**Comment by halocline** [comment]
> Thanks for the feedback, @halocline !
> 
> 1. I think that this could live at the level of `Subheading` as opposed to automatically integrating it into the `Subsection`.

Cool. Let me know how you think we should proceed.

> 2. I think we could apply the same logic to h3s if we end up having those -- I think as of now we just have h1 and h2 for the pages we've put content for. But this could be extended to h3s.

Great.

> 3. On grommet-site, the deep-linking tags items as `property.name`, so that's where I'd pulled `name` from but I am not opposed to using label.

Good to know. I am okay either way.

> 4. The reason I hadn't thought of doing one for h1 (MainHeading) is because there's only going to be one h1 per page, so linking someone to myurl.com/start maybe doesn't need something like myurl.com/start#start. However, I do see the use case if we had a hero component and that heading gets pushed down. Considering these heading components are specific to aries-site, I think we should only add deeplinking to h1 if there is a specific design that necessitates that. From Chris's designs so far though, it seems like the h1 also sits right at the top so it might not be necessary but definitely open to discussing this.

Agree that it is not necessary at this point. I was just running through scenarios I could think of.

**Pull request merged**